packages/grid/x-data-grid/src/hooks/features/rows/useGridRows.ts

```diff
@@ -325,6 +326,57 @@ export const useGridRows = (
     [apiRef, logger],
   );
 
+  const replaceRows = React.useCallback<GridRowApi['replaceRows']>(
+    (firstRowToRender, lastRowToRender, newRows) => {
+      if (props.signature === GridSignature.DataGrid) {
+        throw new Error(
+          [
+            "MUI: You can't replace rows using `apiRef.current.replaceRows` on the DataGrid.",
+            'You need to upgrade to the DataGridPro component to unlock this feature.',
```

Comment:
```suggestion
            'You need to upgrade to DataGridPro or DataGridPremium component to unlock this feature.',
```

To make it consistent with other errors https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid/src/DataGrid/DataGrid.tsx#L103